File: aspnet-core/src/ERP.Healthcare.Application.Contracts/Doctors/CreateUpdateAddressDto.cs
using System.ComponentModel.DataAnnotations;
using Volo.Abp.Application.Dtos;

namespace ERP.Healthcare.Doctors
{
    public class CreateUpdateAddressDto : EntityDto<int>
    {
        [Required]
        [StringLength(100)]
        public string StreetName { get; set; }

        [Required]
        [StringLength(100)]
        public string DistrictName { get; set; }

        [Required]
        public int Postalcode { get; set; }

        [Required]
        public int BuildNumber { get; set; }

        [Required]
        public int ApartmentNumber { get; set; }

        [Required]
        public int CityId { get; set; }

    }
}

File: angular/src/app/doctor/models/create-update-address-dto.ts


import { EntityDto } from '@abp/ng.core';

export class CreateUpdateAddressDto extends EntityDto<number> {
  streetName: string;
  districtName: string;
  postalcode: number;
  buildNumber: number;
  apartmentNumber: number;
  cityId: number;
  id: number;

  constructor(initialValues: Partial<CreateUpdateAddressDto> = {}) {
    super(initialValues);
  }
}
